Automattic/wp-calypso
# Domain Management: Transfer: Add Inter-registrant Transfers

This PR enables the support for inter-registrant transfers.

It changes `/domains/manage/:site/transfer/:domain` to show a menu of options to choose to transfer to another registrar or to another user. This is a bit suboptimal, but not a blocker, and it'll allow us to collect some data on which one is more popular amongst users.

## Screenshots:

### Menu
![screen shot 2016-12-06 at 5 43 39 pm](https://cloud.githubusercontent.com/assets/991022/20951594/93f64966-bbdb-11e6-9256-95cc6609efb8.png)


### Component - Desktop
![screen shot 2016-12-06 at 5 45 58 pm](https://cloud.githubusercontent.com/assets/991022/20951652/e1c2db50-bbdb-11e6-9e49-66fc6b3837c6.png)

### Component - Mobile
![screen shot 2016-12-06 at 5 48 29 pm](https://cloud.githubusercontent.com/assets/991022/20951709/3b0ffe04-bbdc-11e6-8c87-3eefb0287043.png)

### Confirmation - Desktop
![screen shot 2016-12-06 at 5 51 41 pm](https://cloud.githubusercontent.com/assets/991022/20951768/ad68dfac-bbdc-11e6-852a-b2369abf5206.png)


### Confirmation - Mobile
![screen shot 2016-12-06 at 5 50 13 pm](https://cloud.githubusercontent.com/assets/991022/20951742/7a73ab72-bbdc-11e6-9525-7c7ee4e810b7.png)


## Testing

1. Create a secondary testing account, add that user to as an administrator to your site.

2. Go to Domain Management -> Edit -> Transfer -> Transfer to another user and follow instructions.

Verify that it works across providers, with and without privacy protection enabled.

I have tested with all providers with and without privacy. Everything looks OK, but please test it yourselves.

/cc: @aidvu @klimeryk @dzver 

/cc2: @breezyskies would like some design input on this component, but due to its time sensitivity, I will not explicitly wait for a design review to merge this. We can come back to this later on, though.
